### PR TITLE
[webview_flutter] put webview with transparent background

### PR DESCRIPTION
**Pull request by jaumard**
fix https://github.com/flutter/flutter/issues/29300

to use it right now: 

```
webview_flutter: #^0.3.4
    git:
      url: git://github.com/jaumard/plugins
      ref: feature/webview_transparency
      path: packages/webview_flutter
```

### CODE REVIEW

**Comment by nicbabchenko** [comment]
will it be merged soon?

File: packages/webview_flutter/ios/Classes/FlutterWebView.m [hunk 1]
```diff
@@ -263,6 +267,9 @@ - (bool)loadUrl:(NSString*)url {
   }
   NSURLRequest* req = [NSURLRequest requestWithURL:nsUrl];
   [_webView loadRequest:req];
+  _webView.opaque = NO;
```
[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/ios/Classes/FlutterWebView.m]: I remember reading that if you don't on some device or iOS version you might have trouble... but I can't find back the source, maybe I dream it... I'll remove this

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java [hunk 2]
```diff
@@ -101,6 +103,7 @@ public void onMethodCall(MethodCall methodCall, Result result) {
   private void loadUrl(MethodCall methodCall, Result result) {
     String url = (String) methodCall.arguments;
     webView.loadUrl(url);
+    webView.setBackgroundColor(Color.TRANSPARENT);
```
[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: I remember reading that if you don't on some device or iOS version you might have trouble... but I can't find back the source, maybe I dream it... I'll remove this

File: packages/webview_flutter/ios/Classes/FlutterWebView.m [hunk 3]
```diff
@@ -66,6 +66,10 @@ - (instancetype)initWithFrame:(CGRect)frame
     configuration.userContentController = userContentController;
 
     _webView = [[WKWebView alloc] initWithFrame:frame configuration:configuration];
+    _webView.opaque = NO;
```
[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/ios/Classes/FlutterWebView.m]: I didn't find any performance penalty information by playing with this parameter on my side

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: In fact I just tested when update the example app and in fact the web view is not transparent even with this ! :( searching why if anyone have an idea...

[the remark this answers is not in the document]
Comment (amirh) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: I'd guess that the VD hosting it needs to be made transparent as well.

[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: What is VD hosting ? I'm not familiar with that, on iOS side there no problem thing are not working the same in Android?

**Comment by lzxb** [comment]
I've also had problems like yours.
Here's a video

https://github.com/lzxb/flutter-webview-black-screen/blob/master/black-screen.mp4

**Comment by lzxb** [comment]
Flutter webview plugin black screen

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: @amirh bump ^^ :) where do I look for this ?

[the remark this answers is not in the document]
Comment (amirh) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: Getting this to work may require changes to the platform views code in the engine, you can take a look here as a starting point: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

[the remark this answers is not in the document]
Comment (jaumard) [reply packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java]: Thanks @amirh I'll try to take a look at this but I'm going deep into flutter with that lol let's see

**Comment by phamnhuvu-dev** [comment]
Transparent background is not working on Android

**Comment by IchordeDionysos** [comment]
Are there any updates on this? 🤔

**Comment by JasMich** [comment]
Anyone working on?

**Comment by amirh** [comment]
I believe this is possible on Android since https://github.com/flutter/engine/pull/9849 not sure if we even need the setBackgroundColor.

We should probably remove the setBackground calls from everywhere but the constructor. We should also confirm whether the setBackground in the constructor is needed.

This will need a test before it can land.

cc @bparrishMines

**Comment by jaumard** [comment]
@bparrishMines @amirh should be ok now :)

**Comment by TheOneWithTheBraid** [comment]
@amirh What's about this now? I seriously need this change for my `katex_flutter` package. I currently workaround this by using JS to change the HTML background after detecting the BuildContext's background. It would be great if you could merge this.

**Comment by crizant** [comment]
Can't believe it's been a year. Please merge it if it's alright. ;(

**Comment by krispypen** [comment]
I'm also using the fork for some time... please merge 👍

**Comment by crizant** [comment]
Since the original author didn't fix it for months, I decided to create another PR: flutter/plugins#2895

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR (or follow up on https://github.com/flutter/plugins/pull/2895) if you have the time to address the review comments. Thanks!

**Pull request closed without merging**